src/BCFKeyValueData.cc

```diff
@@ -910,6 +935,24 @@ static Status vcf_validate_basic_facts(MetadataCache& metadata,
         return Status::Invalid("gVCF sample names are not unique", dataset + " (" + filename + ")");
     }
 
+    // Make sure we don't have too many contigs
+    int ncontigs = 0;
+    const char **contignames = bcf_hdr_seqnames(hdr, &ncontigs);
+    if (ncontigs > MAX_NUM_CONTIGS_PER_GVCF)
+        return Status::Invalid("Too many contigs (", std::to_string(ncontigs) + ")");
```

Comment:
`Status` wants the first message to be a string literal, so that it doesn't have to be copied. You can set the second message to a `std::string` which will be copied. There are a couple instances where this needs to be changed. Also, all error messages from validating a gVCF file should include the filename. Again, try to imagine being a user and getting the error message- will it help you?